Nos/Router.swift

```diff
@@ -43,6 +47,48 @@ import Dependencies
     func push<D: Hashable>(_ destination: D) {
         currentPath.wrappedValue.append(destination)
     }
+    
+    /// Pushes a view displaying the object wrapped in a `NosNavigationDestination`.
+    func push(_ destination: NosNavigationDestination) {
+        currentPath.wrappedValue.append(destination)
+    }
+    
+    /// Pushes a detail view for the given note.
+    func push(_ note: Event) {
+        push(.note(note.identifier))
+    }
+    
+    /// Pushes a detail view for the note with the given ID, creating one if needed.
+    func push(noteWithID id: RawEventID) {
```

Comment:
The naming doesn't look very swifty to me. I'd prefer something like `pushNote(id: RawEventID)` or even `push(noteID id: RawEventID)` or something similar to that. `push(noteWithID id)` makes me think about a note plus an assumption that the note does have an ID.

Reply:
Yeah good call.